src/gas/GasFeeController.ts

```diff
@@ -426,48 +482,53 @@ export class GasFeeController extends BaseController<
       } else {
         throw new Error('Main gas fee/price estimation failed. Use fallback');
       }
-    } catch {
+    } catch (error) {
+      console.error(error);
       try {
         const estimates = await this.fetchEthGasPriceEstimate(this.ethQuery);
         newState = {
           gasFeeEstimates: estimates,
           estimatedGasFeeTimeBounds: {},
           gasEstimateType: GAS_ESTIMATE_TYPES.ETH_GASPRICE,
         };
-      } catch (error) {
+      } catch (error2) {
         throw new Error(
-          `Gas fee/price estimation failed. Message: ${error.message}`,
+          `Gas fee/price estimation failed. Message: ${error2.message}`,
         );
       }
     }
 
     if (shouldUpdateState) {
-      this.update(() => {
-        return newState;
+      this.update((state) => {
+        state.gasFeeEstimates = newState.gasFeeEstimates;
+        state.estimatedGasFeeTimeBounds = newState.estimatedGasFeeTimeBounds;
+        state.gasEstimateType = newState.gasEstimateType;
       });
     }
 
     return newState;
   }
 
   /**
-   * Remove the poll token, and stop polling if the set of poll tokens is empty.
+   * Removes the given item from the polling queue such that on the next iteration of the poll the
+   * corresponding request for the item will no longer be made.
    *
-   * @param pollToken - The poll token to disconnect.
+   * @param item - The known name of a piece of data for which we are polling.
+   * @deprecated
+   * @see stopPollingFor
    */
-  disconnectPoller(pollToken: string) {
-    this.pollTokens.delete(pollToken);
-    if (this.pollTokens.size === 0) {
-      this.stopPolling();
-    }
+  disconnectPoller(item: keyof PollableItems) {
+    this.stopPollingFor(item);
   }
 
+  /**
+   * Cancels the timer responsible for running the polling queue on a cadence
+   * and removes all items from the polling queue entirely.
+   */
   stopPolling() {
-    if (this.intervalId) {
-      clearInterval(this.intervalId);
-    }
-    this.pollTokens.clear();
-    this.resetState();
+    this.pausePolling();
+    this.pollQueue.clear();
+    // XXX: Previously this called resetState(), do we still want that?
```

Comment:
I really tried to keep the existing behavior with these changes and not introduce backward incompatibilities, but this is one case where I opted to break that. Previously when calling `stopPolling`, we cleared the entire state. We can't do that now since not everything in state has to do strictly with gas fee estimates. What do we want to do here?